src/test/parseCss.test.ts

```diff
@@ -0,0 +1,174 @@
+import { parseCss } from '../parseCss'
+
+import type {} from 'vitest/globals'
+
+test('works', () => {
+    const str = `
+.test {
+    display: flex;
+    |
+}`
+    const pos = str.indexOf('|')
+    const strToParse = str.slice(0, pos) + str.slice(pos + 1)
+    const { usedRules } = parseCss(strToParse, pos)
+    expect(usedRules).toMatchInlineSnapshot(`
+      Map {
+        "display" => {
+          "offset": 13,
+          "value": "flex",
+        },
+      }
+    `)
+})
+test('works with incorrect syntax', () => {
+    const str = `
+.test {
+    display: flex;
+    flex|
+}`
+    const pos = str.indexOf('|')
+    const strToParse = str.slice(0, pos) + str.slice(pos + 1)
+    const { usedRules } = parseCss(strToParse, pos)
+    expect(usedRules).toMatchInlineSnapshot(`
+    Map {
+        "display" => {
+          "offset": 13,
+          "value": "flex",
+        },
+      }
+    `)
+})
+test('throws with incorrent syntax', () => {
+    const str = `
+.test {
+    display: flex;
+    flex|
+}`
+    const pos = str.indexOf('|')
+    const strToParse = str.slice(0, pos) + str.slice(pos + 1)
+    expect(parseCss(strToParse, pos)).not.toThrowErrorMatchingInlineSnapshot(`"expected is not a function"`)
+})
+test('handles template syntax (FirstTemplateToken sytax kind)', () => {
+    const str = `
+const Button = styled.button\`
+  display: flex;
+  flex|    
+\``
+    const pos = str.indexOf('|')
+    const strToParse = str.slice(0, pos) + str.slice(pos + 1)
+
+    const { usedRules } = parseCss(strToParse, pos)
```

Comment:
@zardoy To clarify, do you want me to roll back the implementation of the `src/getTaggedTemplateLangsStylesRange.ts` to 71329a0420dd15d8c3f0a7d3b94f3bbcf669eb85? 

Btw I can't test if it is working right now. It falls with this error and lowering the version of the essential plugins doesn't help (I tried 0.0.42 instead of the latest). Is it the same for you?
![image](https://user-images.githubusercontent.com/74474615/216193961-8584b7c3-b578-4181-a7e5-994fb599f1d8.png)

Reply:
@maIIady Sorry, didn't see your question earlier.

> Btw I can't test if it is working right now.

Isn't that because you're starting in mode with all extensions disabled? You need F5 when no development window is opened. Everything works on my side.

> @zardoy To clarify, do you want me to roll back the implementation of the src/getTaggedTemplateLangsStylesRange.ts to 71329a0?

Not really, I say that for this code:

```ts
css`
    ${'...'}
    display: block;
    flex|
`
```

`parseCss` for now receives whole content css tagged template of including js (`${'...'}`) which is incorrect for js and thats why you handle patching of incorrect css so many times to get rid of css errors.

Instead, here

https://github.com/maIIady/vscode-tailwind-in-css/blob/3e5287ee1d3323e6d60c3e39f5c9242c762a727e/src/getTaggedTemplateLangsStylesRange.ts#L26

pass range of last node instead which will be valid css string and no need to handle fixing it so many times

Reply:
@maIIady You did resolve this, but invalid tests are still here.. Please see start of this thread and code location

Reply:
Didn't find a quick way to implement `getTaggedTemplateStylesRange` testing. Should I use https://www.npmjs.com/package/@vscode/test-electron or is there a simpler way of mocking vscode API? (actually, I struggled to pass `document` to the `getTaggedTemplateStylesRange`)

Reply:
I'll add right now (adopt from vscode-fix-all-json)